src/main/java/org/reactome/addlinks/EnsemblBioMartUtil.java

```diff
@@ -0,0 +1,146 @@
+    public static final String PROTEIN_TO_TRANSCRIPTS_SUFFIX = "_proteinToTranscripts";
+    public static final String TRANSCRIPT_TO_MICROARRAYS_SUFFIX = "_transcriptToMicroarrays";
+    public static final String UNIPROT_TO_PROTEINS_SUFFIX = "_uniprotToProteins";
+
+    /**
+     * Function that takes species name attribute from config file (eg: Homo sapiens) and modifies it to
+     * match Biomart formatting (first letter from genus + full species name, all lowercase -- eg: hsapiens).
+     * @return List<String> of species names in Biomart format (eg: hsapiens).
+     * @throws IOException - Thrown when unable to read file.
+     * @throws ParseException - Thrown when unable to parse JSON data.
+     */
+    public static List<String> getSpeciesNames() throws IOException, ParseException {
+        // Read properties file.
+        Properties applicationProps = getProperties();
+        // Get species JSON config file location, import as JSON object.
+        List<String> speciesNames = getSpeciesNamesFromJSON(applicationProps.getProperty("pathToSpeciesConfig"));
+        return speciesNames;
+    }
+
+    /**
+     * Reads the Species.json file and parses out the species 'name' attribute.
+     * @param pathToSpeciesConfig - String, Filepath to Species.json file
+     * @return List<String> of species names
+     * @throws IOException - Thrown when unable to read file.
+     * @throws ParseException - Thrown when unable to parse JSON data.
+     */
+    private static List<String> getSpeciesNamesFromJSON(String pathToSpeciesConfig) throws IOException, ParseException {
+        JSONParser parser = new JSONParser();
+        Object obj = parser.parse(new FileReader(pathToSpeciesConfig));
+        JSONObject jsonObject = (JSONObject) obj;
+        // For each species in config file, retrieve name, modify it and then add to list.
+        List<String> speciesNames = new ArrayList<>();
+        for (Object speciesKey : jsonObject.keySet()) {
+            JSONObject speciesJson = (JSONObject) jsonObject.get(speciesKey);
+            JSONArray speciesNamesJSON = (JSONArray) speciesJson.get("name");
+            String speciesName = (String) speciesNamesJSON.get(0);
+            speciesNames.add(speciesName);
+        }
+        return speciesNames;
+    }
+
+    /**
+     * Formats species scientific name to BioMart format (eg: hsapiens).
+     * @param speciesName - String, scientific species name (eg: Homo sapiens).
+     * @return String, BioMart-formatted string.
+     */
+    public static String getBioMartSpeciesName(String speciesName) {
+        return speciesName.substring(0,1).toLowerCase() + speciesName.split(" ")[1];
+    }
+
+    /**
+     * Read properties file set to 'config.location' in the system.
+     * @return Loaded properties file
+     * @throws IOException - Thrown if file does not exist.
+     */
+    public static Properties getProperties() throws IOException {
```

Comment:
Does this method need to be public?

Reply:
It is accessed by `EnsemblBioMartRetriever`